kong/plugins/basic-auth/utils.lua

```diff
@@ -0,0 +1,40 @@
+local resty_string = require "resty.string"
+local resty_sha1 = require "resty.sha1"
```

Comment:
There is actually an issue with this. Since we want the encryption to happen at the DAO level, this will be included in the DAO's schema. However, when running unit tests, `resty.string` cannot be required since it is not a module installed by Luarocks, but only available inside of Openresty... It prevents from running unit tests against it but more important, it breaks the `basicauth_credentials` DAO in the unit tests... I am not sure what to do about it. We could either use another module for the encryption or try to mock it somehow.